tests/datasets/test_geo.py

```diff
@@ -152,6 +154,28 @@ def test_and_nongeo(self, dataset: GeoDataset) -> None:
         ):
             dataset & ds2  # type: ignore[operator]
 
+    def test_files_property_warns_non_existing_file(self, tmp_path: Path) -> None:
+        path = (tmp_path / "non_existing_file.tif").as_posix()
+        with pytest.warns(UserWarning, match="Path was ignored."):
+            CustomGeoDataset(paths=path).files
+
+    @pytest.mark.filterwarnings("ignore:.*Path was ignored.*")
+    def test_files_property_for_non_existing_file_or_dir(self, tmp_path: Path) -> None:
```

Comment:
Should be merged. Can we test both a non-existing filename and directory?

Reply:
I though we concluded that these tested two different things, but yes I can merge them. 

It does test both file and dir.

Reply:
I thought we agreed to merge them: https://github.com/microsoft/torchgeo/pull/1612#discussion_r1372262744

The latter tests both but the former only tests file. Just make sure the merged function tests both.

Reply:
You are right, my bad 😅 I totally read that wrong